synapse/handlers/sync.py

```diff
@@ -533,11 +592,23 @@ def compute_state_delta(self, room_id, batch, sync_config, since_token, now_toke
                 state_ids = _calculate_state(
                     timeline_contains=timeline_state,
                     timeline_start=state_at_timeline_start,
+                    timeline_start_members=member_state_ids,
                     previous=state_at_previous_sync,
                     current=current_state_ids,
                 )
             else:
                 state_ids = {}
+                if lazy_load_members:
+                    # TODO: filter out redundant members based on their mxids (not their
```

Comment:
confused. What is a redundant member? Why do they have event_ids? are we currently filtering them by event_ids, or is this a note to remind us not to mistakenly filter by event_ids in the future?

Reply:
have added a comment to answer the above.